SentinelOne/sentinelone_module/singularity/client.py

```diff
@@ -0,0 +1,123 @@
+            # self._rate_limiter = AsyncLimiter(max_rate=1, time_period=3)  # 1 per 3 seconds
+            self._rate_limiter = AsyncLimiter(max_rate=25, time_period=1)  # 25 requests per second
+
+        async with self._rate_limiter:
+            yield self._client
+
+    async def query(self, query: str, variable_values: dict[str, Any] | None = None) -> dict[str, Any]:
+        async with self._session() as session:
+            result: dict[str, Any] = await session.execute_async(gql(query), variable_values=variable_values)
+
+            return result
+
+    async def list_alerts(
+        self, product_name: str, after: str | None = None, start_time: float | None = None
+    ) -> ListAlertsResult:
+        query = """
+            query listAlerts($productName: String!, $after: String, $startDateTime: Long) {
+                alerts(
+                        after: $after,
+                        filters: [
+                          {fieldId: "detectionProduct", stringEqual: { value: $productName }}
+                          {fieldId: "detectedAt", dateTimeRange: { start: $startDateTime, startInclusive: true }}
+                        ]
+                        sort: {by: "detectedAt", order: ASC}
+                ) {
+                    edges {
+                        node {
+                            id
+                            name
+                            description
+                            detectedAt
+                            attackSurfaces
+                            detectionSource{
+                                product
+                            }
+                            name
+                            status
+                            assignee {
+                                fullName
+                            }
+                            classification
+                            confidenceLevel
+                            firstSeenAt
+                            lastSeenAt
+                            process {
+                              cmdLine
+                              file {
+                                path
+                                sha1
+                                sha256
+                                md5
+                              }
+                              parentName
+                            }
+                            result
+                            storylineId
+                        }
+                    },
+                  pageInfo {
+                      endCursor
+                      hasNextPage
+                  }
+                  totalCount
+                }
+            }
+        """
+
+        variables = {"productName": product_name, "after": after, "startDateTime": start_time}
+
+        response = await self.query(query, variable_values=variables)
+
+        if response.get("errors") is not None:
+            raise ValueError(orjson.dumps(response["errors"]).decode("utf-8"))
```

Comment:
I suggest creating a specific exception for this case.
```suggestion
            raise SentinelOneAPIErrors(response["errors"])
```